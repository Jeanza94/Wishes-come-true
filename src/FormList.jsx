
import { useContext } from 'react'
import { FormContext } from './FormContext'
import { useForm } from './hooks/useForm'

export const FormList = () => {

    const { inputValue, handleButton, handleInputValue } = useForm()
    const { handleAddTodo } = useContext(FormContext)

    return (
        <form onSubmit={handleButton} aria-label="form">
            <h3>Your Wish</h3>
            <input
                type="text"
                onChange={handleInputValue}
                value={inputValue}

            />
            <button
                type='submit'
                onClick={() => handleAddTodo(inputValue)}
            >
                New Wish
            </button>
        </form>
    )
}
