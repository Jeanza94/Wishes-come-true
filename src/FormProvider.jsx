import { FormContext } from "./FormContext"
import { useTodo } from "./hooks/useTodo"

export const FormProvider = ({ children }) => {
    const { todos, handleAddTodo, handleRemoveTodo, handleToggleTodo } = useTodo()

    return (
        <FormContext.Provider value={{
            todos,
            handleAddTodo,
            handleRemoveTodo,
            handleToggleTodo
        }}>
            {children}
        </FormContext.Provider>
    )
}
