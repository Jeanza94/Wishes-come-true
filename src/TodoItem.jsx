import { useContext } from "react"
import { FormContext } from "./FormContext"


export const TodoItem = ({ item }) => {

    const { handleRemoveTodo, handleToggleTodo } = useContext(FormContext)

    return (
        <div className="todo-item">
            <li
                aria-label="list"
                className={`todo-item${item.done && "-li"}`}
                onDoubleClick={() => handleToggleTodo(item.id)}
            >
                {item.description}
            </li>
            <button
                onClick={() => handleRemoveTodo(item.id)}
            >
                Good Bye Wish
            </button>
        </div>
    )
}
