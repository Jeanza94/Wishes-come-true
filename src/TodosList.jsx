
import { useContext } from "react"
import { FormContext } from "./FormContext"
import { TodoItem } from "./TodoItem"

export const TodosList = () => {

    const { todos } = useContext(FormContext)

    return (
        <ul>
            {
                todos.map(item => {
                    return (
                        <TodoItem
                            key={item.id}
                            item={item}
                        />
                    )
                })
            }
        </ul>
    )
}
