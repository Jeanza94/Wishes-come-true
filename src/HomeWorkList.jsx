import { FormList } from "./FormList"
import { FormProvider } from "./FormProvider";
import { TodosList } from "./TodosList"

export const HomeWorkList = () => {

    return (
        <FormProvider>
            <h1>Make your wishes come true</h1>
            <p> Let your imagination flies, put your impossible wishes here
                and you will see you are gonna make it.
            </p>
            <div className="list-form">
                <TodosList />
                <FormList />
            </div>
        </FormProvider>



    )
}
