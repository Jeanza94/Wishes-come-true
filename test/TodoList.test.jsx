import { render, screen } from "@testing-library/react";
import { FormContext } from "../src/FormContext";
import { TodosList } from "../src/TodosList";

describe('Pruebas en el TodoList', () => {

    test('Debe de mostrar el componente correctamente', () => {

        const todos = [
            { id: 1, description: 'lavar la losa', done: false },
            { id: 2, description: 'sacar la basura', done: false },
            { id: 3, description: 'planchar la ropa', done: false },
        ]

        const { container } = render(
            <FormContext.Provider value={{ todos: todos }}>
                <TodosList />
            </FormContext.Provider>
        )

        expect(container).toMatchSnapshot();

    });

});