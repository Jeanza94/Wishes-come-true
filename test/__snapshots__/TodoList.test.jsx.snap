// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el TodoList Debe de mostrar el componente correctamente 1`] = `
<div>
  <ul>
    <div
      class="todo-item"
    >
      <li
        aria-label="list"
        class="todo-itemfalse"
      >
        lavar la losa
      </li>
      <button>
        Good Bye Wish
      </button>
    </div>
    <div
      class="todo-item"
    >
      <li
        aria-label="list"
        class="todo-itemfalse"
      >
        sacar la basura
      </li>
      <button>
        Good Bye Wish
      </button>
    </div>
    <div
      class="todo-item"
    >
      <li
        aria-label="list"
        class="todo-itemfalse"
      >
        planchar la ropa
      </li>
      <button>
        Good Bye Wish
      </button>
    </div>
  </ul>
</div>
`;
